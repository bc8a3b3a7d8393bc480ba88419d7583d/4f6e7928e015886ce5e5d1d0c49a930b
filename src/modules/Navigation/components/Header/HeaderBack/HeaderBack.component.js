import React from 'react';
import PropTypes from 'prop-types';
import { withNavigation } from 'react-navigation';
import Icon from 'react-native-vector-icons/Ionicons';

import COLORS from '../../../../../themes/colors';
import HeaderButtonWrapper from '../HeaderButton/HeaderButton.component';

const HeaderBack = ({ navigation: { goBack } }) => (
  <HeaderButtonWrapper onPress={goBack}>
    <Icon name="ios-arrow-round-back" color={COLORS.white} size={30} />
  </HeaderButtonWrapper>
);

HeaderBack.propTypes = {
  navigation: PropTypes.shape({
    goBack: PropTypes.func.isRequired,
  }).isRequired,
};

export { HeaderBack };
export default withNavigation(HeaderBack);
