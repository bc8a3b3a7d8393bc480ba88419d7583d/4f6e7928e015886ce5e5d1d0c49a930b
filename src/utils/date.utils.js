import moment from 'moment';

const supportedDateFormats = {
  ddddDDMMMYYYY: 'dddd, DD MMM YYYY',
};

const format = (date, dateFormat) => moment(date).format(dateFormat);

export default {
  format,
  supportedDateFormats,
};
