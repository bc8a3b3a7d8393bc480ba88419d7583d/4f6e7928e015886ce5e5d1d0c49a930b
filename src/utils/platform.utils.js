import { Platform } from 'react-native';
import DeviceInfo from 'react-native-device-info';
import { equals } from 'ramda';

const isDEV = () => equals(process.env.NODE_ENV, 'development');
const isIOS = () => equals(Platform.OS, 'ios');
const isAndroid = () => equals(Platform.OS, 'android');
const iPhoneX = 'iPhone X';
const isIPhoneX = () => isIOS() && equals(DeviceInfo.getModel(), iPhoneX);

export { isDEV, isIOS, isAndroid, isIPhoneX, iPhoneX };
