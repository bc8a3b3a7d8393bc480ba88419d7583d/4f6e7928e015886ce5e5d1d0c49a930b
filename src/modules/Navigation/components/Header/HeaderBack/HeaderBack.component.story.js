import React from 'react';

// eslint-disable-next-line import/no-extraneous-dependencies
import { storiesOf } from '@storybook/react-native';

import { HeaderBack as Component } from './HeaderBack.component';

storiesOf('Navigation/Component - HeaderBack', module).add('default', () => (
  <Component
    navigation={{
      goBack: () => console.log('go back'),
    }}
  />
));
