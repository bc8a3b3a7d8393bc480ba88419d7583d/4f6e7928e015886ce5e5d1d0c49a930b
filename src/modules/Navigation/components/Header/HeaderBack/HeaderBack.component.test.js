import { testSnapshots } from '../../../../../utils/test.utils';

import { HeaderBack } from './HeaderBack.component';

const props = {
  navigation: {
    goBack: jest.fn(),
  },
};

describe('HeaderBack', () => {
  testSnapshots(HeaderBack, [
    {
      props,
      description: 'basic render',
    },
  ]);
});
