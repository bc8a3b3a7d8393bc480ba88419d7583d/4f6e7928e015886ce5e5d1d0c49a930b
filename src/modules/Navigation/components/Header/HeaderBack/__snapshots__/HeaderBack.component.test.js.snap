// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderBack snapshots basic render 1`] = `
<withNavigation(HeaderButton)
  onPress={[MockFunction]}
>
  <Icon
    allowFontScaling={false}
    color="white"
    name="ios-arrow-round-back"
    size={30}
  />
</withNavigation(HeaderButton)>
`;
